lib/profiles.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+var _ = require('lodash');
+var cacheTimes = require('../web/config/cache-times');
+var fs = require('fs');
+var defaultImage = new Buffer(fs.readFileSync(__dirname + '/../web/public/img/avatar.png', 'base64'), 'base64');
+
+exports.register = function (server, options, next) {
+  options = _.extend({ basePath: '/api/2.0' }, options);
+  var handlers = require('./handlers.js')(server, 'cd-profiles');
+
+  server.route([{
+    method: 'POST',
+    path: options.basePath + '/profiles/user-profile-data',
+    handler: handlers.actHandler('user_profile_data')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/create',
+    handler: handlers.actHandlerNeedsUser('create')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/youth/create',
+    handler: handlers.actHandlerNeedsUser('save-youth-profile')
+  }, {
+    method: 'PUT',
+    path: options.basePath + '/profiles/youth/update',
+    handler: handlers.actHandlerNeedsUser('update-youth-profile')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/invite-parent-guardian',
+    handler: handlers.actHandlerNeedsUser('invite-parent-guardian')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/accept-parent-guardian',
+    handler: handlers.actHandlerNeedsUser('accept-parent-invite')
```

Comment:
ah - no, that's what they are, it's very inconsistent

On Wed, Oct 14, 2015 at 8:27 PM, Cristian Kiss <notifications@github.com>
wrote:

> In lib/profiles.js
> <https://github.com/CoderDojo/cp-zen-platform/pull/703#discussion_r42040712>
> :
>
> > +  }, {
> > +    method: 'POST',
> > +    path: options.basePath + '/profiles/youth/create',
> > +    handler: handlers.actHandlerNeedsUser('save-youth-profile')
> > +  }, {
> > +    method: 'PUT',
> > +    path: options.basePath + '/profiles/youth/update',
> > +    handler: handlers.actHandlerNeedsUser('update-youth-profile')
> > +  }, {
> > +    method: 'POST',
> > +    path: options.basePath + '/profiles/invite-parent-guardian',
> > +    handler: handlers.actHandlerNeedsUser('invite-parent-guardian')
> > +  }, {
> > +    method: 'POST',
> > +    path: options.basePath + '/profiles/accept-parent-guardian',
> > +    handler: handlers.actHandlerNeedsUser('accept-parent-invite')
>
> sorry @dberesford <https://github.com/dberesford> I meant handler names
> should use underscore, like accept_parent_invite :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CoderDojo/cp-zen-platform/pull/703/files#r42040712>.
>